nupic/research/frameworks/pytorch/model_utils.py

```diff
@@ -103,16 +106,21 @@ def train_model(
         loss = criterion(output, target)
         del data, target, output
 
+        t2 = time.time()
         if use_amp:
             with amp.scale_loss(loss, optimizer) as scaled_loss:
                 scaled_loss.backward()
         else:
             loss.backward()
 
+        t3 = time.time()
         optimizer.step()
+        t4 = time.time()
 
         if post_batch_callback is not None:
-            post_batch_callback(model=model, loss=loss.detach(), batch_idx=batch_idx)
+            post_batch_callback(model=model, loss=loss.detach(), batch_idx=batch_idx,
```

Comment:
Have you tried using the "profile=True" in your config?

Reply:
I'm running with -p on a regular basis now.

Reply:
'-p' is for showing progress. "profile=True" uses pytorch profiler to show timing and other profiling information